homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -351,8 +351,9 @@ def __init__(self, config):
 
         if HUE_API_STATE_BRI in request_json:
             if entity.domain == light.DOMAIN:
-                parsed[STATE_ON] = parsed[STATE_BRIGHTNESS] > 0
-                if not entity_features & SUPPORT_BRIGHTNESS:
+                if entity_features & SUPPORT_BRIGHTNESS:
+                    parsed[STATE_ON] = parsed[STATE_BRIGHTNESS] > 0
+                else:
```

Comment:
This can be rewritten to remove the `else:`. It is unneeded here.

Reply:
Can you elaborate on what you mean? Those two actions are mutually exclusive, so quite literally, `else` has to be there in one form or another unless we want to do bigger refactor here. The goal here was to fix only this one issue without amending the behavior in other cases.

Reply:
Also, I wanted to add a test case for this, but since I never saw this code before, I'm struggling with that. The entity "added" in test suite for this component called `light.no_brightness` is not really a light. It does not react to service `turn_on` and `turn_off` calls - this makes me unsure on how to verify that entities' state has been correctly amended. I did manage to get working test, but only by adding one more light to `demo` platform setup, which doesn't feel right. 
Any tips on how to approach it?

Reply:
Aah I see, please remind me to drink coffee before reviewing in the morning 😉

Reply:
I've extended tests by using service mock, so that I don't need an actual working light, but I'm still not sure whether this is a way to go. The test would be much cleaner if this entity was a working light, but I'm still not sure how to easily make one, unless adding entities to `demo` platform is something that is "OK".